nixos/modules/services/misc/filebrowser.nix

```diff
@@ -0,0 +1,89 @@
+{ config, pkgs, lib, ... }:
+let cfg = config.services.filebrowser;
+in {
+  options = {
+    services.filebrowser = {
+      dataDir = lib.mkOption {
+        type = lib.types.str;
+        default = "/var/lib/filebrowser";
+        description = "Directory where Filebrowser stores its data files";
+      };
+      enable = lib.mkEnableOption "Filebrowser";
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = "filebrowser";
+        description = "Group under which Filebrowser runs";
+      };
+      listenAddress = lib.mkOption {
+        type = lib.types.str;
+        default = "0.0.0.0";
+        description = "Address that Filebrowser will bind to";
+      };
+      listenPort = lib.mkOption {
+        type = lib.types.port;
+        default = 5983;
+        description = "Port that Filebrowser will bind to";
+      };
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Open ports in firewall for Filebrowser web interface";
+      };
+      package = lib.mkPackageOption pkgs "filebrowser" { };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "filebrowser";
+        description = "User under which Filebrowser runs";
+      };
```

Comment:
I would reorder the optins like this.
This way `enable` and `package` are at the top and `user` and `group` are next to each other.
```suggestion
      enable = lib.mkEnableOption "Filebrowser";
      package = lib.mkPackageOption pkgs "filebrowser" { };
      dataDir = lib.mkOption {
        type = lib.types.str;
        default = "/var/lib/filebrowser";
        description = "Directory where Filebrowser stores its data files";
      };
      user = lib.mkOption {
        type = lib.types.str;
        default = "filebrowser";
        description = "User under which Filebrowser runs";
      };
      group = lib.mkOption {
        type = lib.types.str;
        default = "filebrowser";
        description = "Group under which Filebrowser runs";
      };
      listenAddress = lib.mkOption {
        type = lib.types.str;
        default = "0.0.0.0";
        description = "Address that Filebrowser will bind to";
      };
      listenPort = lib.mkOption {
        type = lib.types.port;
        default = 5983;
        description = "Port that Filebrowser will bind to";
      };
      openFirewall = lib.mkOption {
        type = lib.types.bool;
        default = false;
        description = "Open ports in firewall for Filebrowser web interface";
      };
```

Reply:
sounds good. I usually default to trying to put stuff in lexical order when possible but I'll take the advice :+1: